apiary-metastore-events/sns-metastore-events/apiary-metastore-listener/src/main/java/com/expediagroup/apiary/extensions/events/metastore/listener/ApiarySnsListener.java

```diff
@@ -134,7 +134,21 @@ public void onInsert(InsertEvent event) throws MetaException {
     if (event.getStatus() == false) {
       return;
     }
-    publishInsertEvent(EventType.INSERT, event.getDb(), event.getTable(), event.getPartitionKeyValues(),
+
+    Table table = event.getTableObj();
+    String databaseName = table.getDbName();
+    String tableName = table.getTableName();
+
+    LinkedHashMap<String, String> partitionKeyValues =  new LinkedHashMap<>();
+    if (event.getPartitionObj() != null) {
```

Comment:
If there is a good common place this could be refactored out into (since we do this twice from what I can see) that would be nice, but if there isn't something immediately obvious then let's just leave it like this for now.

Reply:
Took a shot at it by making it a static method on ApiaryInsertEvent and including apiary-hive-events as a dependency for sns-metastore-listener.